### ISSUE TITLE
isset can not exchanged by !== everytime

### ISSUE THREAD

**Issue by dpauli** [kalessil/phpinspectionsea#301]
There is an annoying hint on `if` clauses:

```php
<?php
if (isset($var)) {
   // make some Stuff
}
```

The suggestion is to use

```php
if (null !== $var) {

}
```

This throws a notice, if `$var` is never set.

**Comment by kalessil** [comment]
Thank you for reporting @dpauli, this can be disabled (a check box bottom-right): 
![image](https://cloud.githubusercontent.com/assets/1577185/25801664/e28d4986-33ee-11e7-9ca2-0d64a7314da6.png)

But it's can be a false-positive as well. Can you provide more info (ideally OSS link or complete template/function/method)?

**Comment by kalessil** [comment]
@dpauli : any chance to get code samples for analysis?

**Comment by dpauli** [comment]
Sorry for late response, There is some code where this problem appears:

```php
if (isset($neverInitializedVar)) {
    echo 'someStuff';
}
```

In this case I get the following suggestion, see image.

![nullfalsepositive](https://cloud.githubusercontent.com/assets/10072693/25938652/3e7224e0-3630-11e7-8874-7cbe90a7fea0.png)

I cant use this suggestion `null !==` because this variable is never set before.

I have the latest PHPStorm (2017.1.3, because of EAP), but it also works wrong on open released versions (2017.1.2). Work on Windows and have the latest plugin.

**Comment by rentalhost** [comment]
Seems valid on this context. My suggestion is check if the scope is closed (like function), then `isset($var)` should be not allowed by checking firstly that `$var` doesn't exists. In global scope seems to be a valid call when is not possible check if `$var` exists. If it exists, then it should works with the current implementation.

**Comment by funivan** [comment]
My use case: for example we have `view` files (aka php, html ). This files can not be recognized by phpstorm like views in twig. They are just plain php files with global scope. 
But we can create own scope in phpstorm and change severity of this inspection in the specific scope.
Maybe it will helps someone 
**Configure scope**
![10 46 34](https://cloud.githubusercontent.com/assets/425208/25988449/b40b1538-3700-11e7-8138-11b4d62b3f4e.png)

**Configure inspection for the specific scope**
![10 46 02](https://cloud.githubusercontent.com/assets/425208/25988467/c2d16fd6-3700-11e7-8cbd-d8b5d4c60efd.png)

**Comment by kalessil** [comment]
I don't know, I should probably limit the inspection to methods/functions.

**Comment by kalessil** [comment]
Variables in the global context are not checked anymore.
